etl/steps/data/garden/terrorism/2023-07-20/global_terrorism_database.py

```diff
@@ -0,0 +1,253 @@
+    df_pop_deaths["terrorism_deaths_per_capita"] = df_pop_deaths["total_killed"] / df_pop_deaths["population"]
+    df_pop_deaths["terrorism_casualties_per_capita"] = df_pop_deaths["total_casualties"] / df_pop_deaths["population"]
+
+    # Perform decadal averaging for selected columns
+    cols_for_decadal_av = [
+        "total_killed",
+        "total_wounded",
+        "total_incident_counts",
+        "total_casualties",
+        "terrorism_wounded_per_capita",
+        "terrorism_deaths_per_capita",
+        "terrorism_casualties_per_capita",
+    ]
+
+    df_pop_deaths = perform_decadal_averaging(df_pop_deaths, cols_for_decadal_av=cols_for_decadal_av)
+
+    # Convert relevant columns to float64 data type (to avoid errors related to this issue -  https://github.com/owid/etl/issues/1334)
+    df_pop_deaths["terrorism_casualties_per_capita"] = df_pop_deaths["terrorism_casualties_per_capita"].astype(
+        "float64"
+    )
+    df_pop_deaths["terrorism_deaths_per_capita"] = df_pop_deaths["terrorism_deaths_per_capita"].astype("float64")
+    df_pop_deaths["terrorism_wounded_per_capita"] = df_pop_deaths["terrorism_wounded_per_capita"].astype("float64")
+
+    # Drop total deaths and population columns
+    df_pop_deaths.drop(["deaths", "population"], axis=1, inplace=True)
+    # Convert DataFrame to a new garden dataset table.
+    tb_garden = Table(df_pop_deaths, short_name=paths.short_name)
+    tb_garden.set_index(["country", "year"], inplace=True)
+
+    # Create a new garden dataset with the same metadata as the meadow dataset.
+    ds_garden = create_dataset(dest_dir, tables=[tb_garden], default_metadata=ds_meadow_terrorism.metadata)
+
+    # Save changes in the new garden dataset.
+    ds_garden.save()
+
+
+def perform_decadal_averaging(df: pd.DataFrame, cols_for_decadal_av: list) -> pd.DataFrame:
+    """
+    Perform decadal averaging on the specified columns in the DataFrame.
+
+    This function calculates the mean value for each specified column over the decades. It groups the data by decade
+    based on the 'year' column and computes the mean for each decadal group. Non-decadal years are replaced with NaN.
+
+    Parameters:
+        df (pd.DataFrame): The DataFrame containing the data.
+        cols_for_decadal_av (list): A list of column names for which decadal averaging should be performed.
+
+    Returns:
+        pd.DataFrame: A modified DataFrame with additional columns representing decadal averages for the specified
+                      columns.
+    """
+
+    for column in df[cols_for_decadal_av]:
+        # Create a new column name for decadal average
+        decadal_column = f"decadal_{column}"
+
+        # Group the data by decade and compute the mean for each decadal group
+        df[decadal_column] = df.groupby(df["year"] // 10 * 10)[column].transform("mean")
+
+        # Replace non-decadal years with NaN
+        df[decadal_column] = df[decadal_column].mask(df["year"] % 10 != 0, pd.NA)
```

Comment:
If you decide to work with `float` (see my comment above), then you should use `np.nan` instead of `pd.NA`. I'd stay away from `pd.NA` until we upgrade to pandas 2.0.